File: src/main/java/net/dontdrinkandroot/gitki/wicket/component/item/SignInPageLinkItem.java
package net.dontdrinkandroot.gitki.wicket.component.item;

import net.dontdrinkandroot.gitki.wicket.GitkiWebSession;
import net.dontdrinkandroot.gitki.wicket.page.SignInPage;
import net.dontdrinkandroot.wicket.bootstrap.component.item.BookmarkablePageLinkItem;
import org.apache.wicket.Page;
import org.apache.wicket.model.ResourceModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class SignInPageLinkItem extends BookmarkablePageLinkItem
{
    public <C extends Page> SignInPageLinkItem(String id)
    {
        super(id, new ResourceModel("gitki.signin"), SignInPage.class);
    }

    @Override
    protected void onConfigure()
    {
        super.onConfigure();
        this.setVersioned(false);
        if (null != GitkiWebSession.get().getUser()) {
            this.setVisible(false);
        }
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/component/item/SignInPageLinkItem.kt
package net.dontdrinkandroot.gitki.wicket.component.item

import net.dontdrinkandroot.gitki.wicket.getCurrentUser
import net.dontdrinkandroot.gitki.wicket.page.SignInPage
import net.dontdrinkandroot.wicket.bootstrap.component.item.BookmarkablePageLinkItem
import org.apache.wicket.model.ResourceModel

class SignInPageLinkItem(id: String) :
    BookmarkablePageLinkItem<Any>(id, label = ResourceModel("gitki.signin"), pageClass = SignInPage::class.java) {

    override fun onConfigure() {
        super.onConfigure()
        this.isVersioned = false
        if (null != getCurrentUser()) {
            this.isVisible = false
        }
    }
}